### PR TITLE
FW position control add takeoff minimum pitch

### PR DESCRIPTION
**Pull request by dagar**
This is currently passed through via mavlink, but I don't think it makes sense to set per mission. This is a vehicle parameter.

 - fixes https://github.com/PX4/Firmware/issues/6787

### CODE REVIEW

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 1]
```diff
@@ -1309,7 +1309,7 @@ FixedwingPositionControl::control_takeoff(const Vector2f &curr_pos, const Vector
 					   _parameters.throttle_max, // XXX should we also set runway_takeoff_throttle here?
 					   _parameters.throttle_cruise,
 					   _runway_takeoff.climbout(),
-					   radians(_runway_takeoff.getMinPitch(pos_sp_curr.pitch_min, 10.0f, _parameters.pitch_limit_min)),
```
Comment (Antiheavy) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp]: comment only: additional benefit is removal of hidden hard-coded values like this 10 degree thing...

File: src/modules/mavlink/mavlink_mission.cpp [hunk 2]
```diff
@@ -1309,13 +1309,6 @@ MavlinkMissionManager::parse_mavlink_mission_item(const mavlink_mission_item_t *
 			mission_item->altitude_is_relative = true;
 		}
 
-		/* this field is shared with pitch_min (and circle_radius for geofence) in memory and
-		 * exclusive in the MAVLink spec. Set it to 0 first
-		 * and then set minimum pitch later only for the
-		 * corresponding item
-		 */
-		mission_item->time_inside = 0.0f;
```
Comment (Antiheavy) [inline src/modules/mavlink/mavlink_mission.cpp]: is this related to FW_TKO_PITCH_MIN or is this extra cleanup?

**Comment by dagar** [comment]
Testing this in SITL is actually a bit difficult.

**Comment by Antiheavy** [comment]
One thing I am still confused about:  will this takeoff minimum pitch be valid until FW_CLMBOUT_DIFF or until MIS_TAKEOFF_ALT?  or is it both (essentially whichever is greater)?

**Comment by Antiheavy** [comment]
> Testing this in SITL is actually a bit difficult.

can you put an unreasonable value into QGC's Takeoff Waypoint dialog and verify bad things happen without this PR and that okay thing happen with this PR, i.e. the value is ignored?  I don't have my SITL set up with me today, but can try tomorrow.
image

**Comment by Antiheavy** [comment]
> Testing this in SITL is actually a bit difficult.

actually you are right, this is hard to test in SITL.  I don't know if it is due to the way the SITL model works, or if there is actually something else going on.  I can't get the vehicle to behave differently during takeoff no matter what "Pitch" value I place in the TakeoffWP param.  This doesn't match our experience in real flight test.  Might have to go do some real flight tests on this.

**Comment by dagar** [comment]
Testing as part of https://github.com/PX4/Firmware/pull/11845

**Comment by Antiheavy** [comment]
We finally got out to flight test this PR.  It works!  It was tested on a lightly customized branch of 1.8.x firmware so it may need re-testing as part of 1.9.x, but the logic works.

Fixed wing flight log:
https://review.px4.io/plot_app?log=68ef0917-824e-446b-9aff-dc8e93fcf64c

4 short flights with FW_TKO_PITCH_MIN set to different values.  You can see the vehicle's takeoff and climbout pitch behavior changes according to the parameter values.  It is also successfully ignoring the default value set in QGC (15 degrees).
image

**Comment by LorenzMeier** [comment]
Closing this in favor of #11845

**Pull request closed without merging**